### PR TITLE
feat: add debug for snyk request id header

### PR DESCRIPTION
**Pull request by DoronElir**
- [ ] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/CONTRIBUTING.md) rules

#### What does this PR do?
Add debug log for `snyk-request-id` on monitor request

#### Any background context you want to provide?
This is needed for debugging some support tickets where the `snyk-request-id` is needed

### CODE REVIEW

File: src/lib/monitor/index.ts [hunk 1]
```diff
@@ -268,6 +268,10 @@ async function monitorDepTree(
     if (!userMessage && res.statusCode === 504) {
       throw new ConnectionTimeoutError();
     } else {
+      debug(
+        'the header snyk-request-id is' + res.headers['snyk-request-id'] ??
+          'undefined',
+      );
```
Comment (gitphill) [inline src/lib/monitor/index.ts:274]: is it not also interesting for connection timeout error (above)?

Comment (gitphill) [reply src/lib/monitor/index.ts:274]: i think `monitorDepGraph` is really where you want this though

Comment (DoronElir) [reply src/lib/monitor/index.ts:274]: Absolutely, thanks for the comment 🤟

Comment (gitphill) [reply src/lib/monitor/index.ts:274]: or within `makeRequest` itself?

Comment (DoronElir) [reply src/lib/monitor/index.ts]: not agreeing there, we are not the only one who is using `makeRequest ` and maybe it should be there, but I think it makes this change much bigger

**Pull request closed without merging**